R/teal_data_module.R

```diff
@@ -81,118 +81,3 @@ teal_data_module <- function(ui, server, label = "data module", once = TRUE) {
-#' @examples
-#' my_transformers <- list(
-#'   teal_transform_module(
-#'     label = "Custom transform for iris",
-#'     datanames = "iris",
-#'     ui = function(id) {
-#'       ns <- NS(id)
-#'       tags$div(
-#'         numericInput(ns("n_rows"), "Subset n rows", value = 6, min = 1, max = 150, step = 1)
-#'       )
-#'     },
-#'     server = function(id, data) {
-#'       moduleServer(id, function(input, output, session) {
-#'         reactive({
-#'           within(data(),
-#'             {
-#'               iris <- head(iris, num_rows)
-#'             },
-#'             num_rows = input$n_rows
-#'           )
-#'         })
-#'       })
-#'     }
-#'   )
-#' )
-#'
-#' @name teal_transform_module
-#'
-#' @export
-teal_transform_module <- function(ui = function(id) NULL,
-                                  server = function(id, data) data,
-                                  label = "transform module",
-                                  datanames = character(0)) {
-  checkmate::assert_function(ui, args = "id", nargs = 1)
-  checkmate::assert_function(server, args = c("id", "data"), nargs = 2)
-  checkmate::assert_string(label)
-  checkmate::assert_character(datanames)
-  if (identical(datanames, "all")) {
-    stop(
-      "teal_transform_module can't have datanames property equal to 'all'. Set `datanames = character(0)` instead.",
-      call. = FALSE
-    )
-  }
-  structure(
-    list(
-      ui = ui,
-      server = function(id, data) {
-        data_out <- server(id, data)
-
-        if (inherits(data_out, "reactive.event")) {
-          # This warning message partially detects when `eventReactive` is used in `data_module`.
-          warning(
-            "teal_transform_module() ",
-            "Using eventReactive in teal_transform module server code should be avoided as it ",
-            "may lead to unexpected behavior. See the vignettes for more information  ",
-            "(`vignette(\"data-transform-as-shiny-module\", package = \"teal\")`).",
-            call. = FALSE
-          )
-        }
-
-        decorate_err_msg(
-          assert_reactive(data_out),
-          pre = sprintf("From: 'teal_transform_module()':\nA 'teal_transform_module' with \"%s\" label:", label),
-          post = "Please make sure that this module returns a 'reactive` object containing 'teal_data' class of object." # nolint: line_length_linter.
-        )
-      }
-    ),
-    label = label,
-    datanames = datanames,
-    class = c("teal_transform_module", "teal_data_module")
-  )
-}
-
-
-#' Extract all `transformers` from `modules`.
-#'
-#' @param modules `teal_modules` or `teal_module`
-#' @return A list of `teal_transform_module` nested in the same way as input `modules`.
-#' @keywords internal
-extract_transformers <- function(modules) {
-  if (inherits(modules, "teal_module")) {
-    modules$transformers
-  } else if (inherits(modules, "teal_modules")) {
-    lapply(modules$children, extract_transformers)
-  }
-}
```

Comment:
`extract_transformers` is used in init https://github.com/insightsengineering/teal/blob/main/R/init.R#L215

Reply:
This is responsible for failures of 4 tests - bringing back for now

Reply:
Ah, this was moved to teal_transform_module - but the name was changed. The name needs to be changed in `init` as well. `extract_transformers` -> `extract_transforms`